inspirehep/modules/workflows/workflows/article.py

```diff
@@ -492,8 +495,8 @@ class Article(object):
         NOTIFY_IF_SUBMISSION +
         MARK_IF_MATCH_IN_HOLDINGPEN +
         CHECK_IS_UPDATE +
-        PROCESS_HOLDINGPEN_MATCHES +
         ENHANCE_RECORD +
+        PROCESS_HOLDINGPEN_MATCHES +
```

Comment:
not sure about performance implications of fully enhancing all records before matching. @ammirate @david-caro ? otherwise you could just pull the required step in front, in say `PRE_ENHANCE_RECORD`.

Reply:
It's a quite big load, though we don't have really much data to back that comment up (yet).

Reply:
We can actually move the 'auto-approve' step from the `ENHANCE_RECORD`, to it's own, and put that one only before the `PROCESS_HOLDINGPEN_MATCHES`, as it does not really need anything from the enhanced info.

Reply:
this `IF_ELSE` block:
https://github.com/inspirehep/inspire-next/pull/3258/files#diff-16f31546a6d82e399e344031e67a88c6R158

Reply:
on it